Changed the icon color for the Magic tone of the Badge component. (#11401)
Shopify/polaris: .changeset/tasty-doors-arrive.md

---

> '@shopify/polaris': minor

Reviewer: ```suggestion
'@shopify/polaris': patch
```

---

> Changed the icon color from icon-magic to text-magic for the Magic tone of the Badge component.

Reviewer: ```suggestion
Aligned the magic `Badge` `tone` icon and text colors 
```
